pkg/scc/update.go

```diff
@@ -19,54 +19,184 @@ limitations under the License.
+	Config_submariner_operator_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-operator-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-operator
+  namespace: submariner-operator
+`
+	Config_submariner_routeagent_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-routeagent-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-routeagent
+  namespace: submariner-operator
+`
+	Config_submariner_globalnet_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-globalnet-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-globalnet
+  namespace: submariner-operator
+`
+	Config_submariner_networkplugin_syncer_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-networkplugin-syncer-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-networkplugin-syncer
+  namespace: submariner-operator
+`
+
+	Config_submariner_diagnose_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-diagnose-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-diagnose
+  namespace: submariner-operator
+`
+	Config_submariner_lighthouse_agent_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-lighthouse-agent-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-lighthouse-agent
+  namespace: submariner-operator
+`
+
+	Config_submariner_lighthouse_coredns_privilged_scc_clusterrolebinding_yaml = `---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: submariner-lighthouse-coredns-privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: submariner-privileged-scc
+subjects:
+- kind: ServiceAccount
+  name: submariner-lighthouse-coredns
+  namespace: submariner-operator
+`
+)
+
+var (
+	privilegedClusterRoleDefined = false
```

Comment:
Why is this defined globally? We really should avoid mutable global state.

Reply:
To avoid calling ensure function if ClusterRole resource is already created, I guess we can delete this check.

Reply:
Since the ClusterRole only need be created once, perhaps we can let the caller do it in the appropriate place.